### PR TITLE
ComboBox: fix input focus stealing issues

### PR DESCRIPTION
**Pull request by jspurlin**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [X] Include a change request file using `$ npm run change`

#### Description of changes

Before this change the comboBox would set focus back to the input after the menu was closed which could cause problems if the creator was setting focus when processing the new input from the menu.

The fix makes sure we are only setting focus back to the input when needed (pretty much if the menu was dropped via keyboard, if dropped via mouse we should not force focus when the menu is closed)

#### Focus areas to test

Verified that we are only setting focus back to the input when need and are not making focus/select calls if the menu is closed after being opened via mouse. Also, non-expanded behavior is not affected

### CODE REVIEW

**Comment by christiango** [comment]
Build failures are weird, can you pull from the upstream repo to queue a new build? They seem unrelated to your changes

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -1363,6 +1378,7 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
         // make space expand/collapse the comboBox
         // and allow the event to propagate
         if (!allowFreeform && autoComplete === 'off') {
+          this._focusInputAfterClose = !!this.state.isOpen;
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Does it make sense to not call setState directly, but have a helper function that calls set state and also sets this._focusInputAfterClose?

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 2]
```diff
@@ -131,6 +131,15 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
 
   private _scrollIdleTimeoutId: number | undefined;
 
+  // This tracks how the comboBox was opened
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Comment is a little misleading, it doesn't actually track anything about how the combo box was opened, it just says what should happen after close.

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 3]
```diff
@@ -203,24 +212,24 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
       this._async.setTimeout(() => this._scrollIntoView(), 0);
     }
 
-    // If we are open or we are focused but are not the activeElement,
-    // set focus on the input
-    if (isOpen || (focused && document.activeElement !== this._comboBox.inputElement)) {
+    // If we are open or we are just closed, should focusAfter close, are focused but
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: nit: replace should focusAfter close with shouldFocusAfterClose is set

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 2 above)
Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: true, I'll update the comment

**Comment by christiango** [comment]
@battletoilet this is another PR that is failing because of the build breaks from 2c399b1a3ddf10b8707e66ec53f3f1bbaf3d0a65

**Comment by jspurlin** [comment]
@christiango Thanks for investigating the experiments failure root cause

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 3 above)
Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: done

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Yeah, good call, it also helps make it more clear when we are updating that value

**Comment by jspurlin** [comment]
Also, screener is happy: 
image

**Pull request merged**